packages/web/src/components/cells/code.tsx

```diff
@@ -729,17 +782,40 @@ function CodeEditor({
     extensions.push(EditorView.editable.of(false));
     extensions.push(EditorState.readOnly.of(true));
   }
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalContent, setModalContent] = useState('');
+
+  const openModal = (content: string) => {
+    setModalContent(content);
+    setIsModalOpen(true);
+  };
 
   return (
-    <CodeMirror
-      value={cell.source}
-      theme={codeTheme}
-      extensions={extensions}
-      onChange={(source) => {
-        updateCellOnClient({ ...cell, source });
-        updateCellOnServerDebounced(cell, { source });
-      }}
-    />
+    <>
+      <CodeMirror
+        value={cell.source}
+        theme={codeTheme}
+        extensions={extensions}
+        onChange={(source) => {
+          updateCellOnClient({ ...cell, source });
+          updateCellOnServerDebounced(cell, { source });
+        }}
+      />
+      <Dialog open={isModalOpen} onOpenChange={setIsModalOpen}>
```

Comment:
Fine to merge with this but in general it is probably cleaner to have one dialog rendered for this purpose rather than N.